src/allmydata/web/common.py

```diff
@@ -480,8 +500,38 @@ def render(self, req):
                 req.setResponseCode(e.code)
                 return json.dumps({"error": e.text})
             except Exception as e:
-                message, code = humanize_failure(Failure())
+                message, code = humanize_exception(e)
                 req.setResponseCode(500 if code is None else code)
                 return json.dumps({"error": message})
         else:
             raise WebError("'%s' invalid type for 't' arg" % (t,), http.BAD_REQUEST)
+
+
+def exception_to_child(f):
+    """
+    Decorate ``getChild`` method with exception handling behavior to render an
+    error page reflecting the exception.
+    """
+    @wraps(f)
+    def g(self, name, req):
+        try:
+            return f(self, name, req)
+        except Exception as e:
+            description, status = humanize_exception(e)
+            return resource.ErrorPage(status, "Error", description)
+    return g
+
+
+def render_exception(f):
+    """
+    Decorate a ``render_*`` method with exception handling behavior to render
+    an error page reflecting the exception.
+    """
+    @wraps(f)
+    def g(self, request):
+        try:
+            return f(self, request)
+        except Exception as e:
+            description, status = humanize_exception(e)
+            return resource.ErrorPage(status, "Error", description).render(request)
+    return g
```

Comment:
I quite like this.  This is way better than my pedestrian way of adding exception handlers in every `Resource` instance. :-) :+1:

Reply:
I started thinking about adding those exception handlers and quickly realized it would be too boring to actually completely follow through on it. :)